allennlp/common/params.py

```diff
@@ -45,9 +45,40 @@ class Params(MutableMapping):
     # and passing no value to the default parameter of "pop".
     DEFAULT = object()
 
-    def __init__(self, params: Dict[str, Any], history: str = "") -> None:
+    def __init__(self,
+                 params: Dict[str, Any],
+                 history: str = "",
+                 loading_from_archive: bool = False,
+                 files_to_archive: Dict[str, str] = None) -> None:
         self.params = _replace_none(params)
         self.history = history
+        self.loading_from_archive = loading_from_archive
+        self.files_to_archive = {} if files_to_archive is None else files_to_archive
```

Comment:
How do you feel about `self.files_to_archive = files_to_archive or {}`?

Reply:
I had it that way originally, but that creates a bug. basically, we want `files_to_archive` shared down through the children, but if you write it that way then when the parent has an empty dict the child gets a *different* empty dict

Reply:
Ah, great, good point.